### PR TITLE
Convert external chunking documentation to doxygen

### PR DESCRIPTION
**Pull request by byrnHDF**

### CODE REVIEW

**Review by lrknox — DISMISSED** [review]

File: doxygen/dox/DSChunkingIssues.dox [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+ *
+ * <table>
+ * <tr>
+ * <td>
+ * \image html Chunk_f1.gif "Figure 1"
+ * </td>
+ * </tr>
+ * </table>
+ *
+ * The HDF5 library treats chunks as atomic objects -- disk I/O is always in terms of complete chunks (Parallel versions
```
Comment (derobins) [inline doxygen/dox/DSChunkingIssues.dox]: lower case p in Parallel

File: doxygen/dox/DSChunkingIssues.dox [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+ * <td>
+ * \image html Chunk_f2.gif "Figure 2"
+ * </td>
+ * </tr>
+ * </table>
+ *
+ * \section sec_hdf5_chunk_issues_data The Raw Data Chunk Cache
+ * It's obvious from Figure 2 that calling #H5Dwrite many times from the application would result in poor performance even
+ * if the data being written all falls within a single chunk. A raw data chunk cache layer was added between the top of
+ * the filter stack and the bottom of the byte modification layer (The raw data chunk cache was added before the second alpha release.).
```
Comment (derobins) [inline doxygen/dox/DSChunkingIssues.dox]: The comment in parens is obsolete and should be removed.

File: doxygen/dox/DSChunkingIssues.dox [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+ * By default, the chunk cache will store 521 chunks
+ * or 1MB of data (whichever is less) but these values can be modified with #H5Pset_cache.
+ *
+ * The preemption policy for the cache favors certain chunks and tries not to preempt them.
+ * \li Chunks that have been accessed frequently in the near past are favored.
+ * \li A chunk which has just entered the cache is favored.
+ * \li A chunk which has been completely read or completely written but not partially read or written is penalized according
+ * to some application specified weighting between zero and one.
+ * \li A chunk which is larger than the maximum cache size is not eligible for caching.
```
Comment (derobins) [inline doxygen/dox/DSChunkingIssues.dox:46]: @fortnern - Does this section need revising?

File: doxygen/dox/DSChunkingIssues.dox [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+ *
+ * <table>
+ * <tr>
+ * <td>
+ * \image html Chunk_f5.gif "Figure 5"
+ * </td>
+ * </tr>
+ * </table>
+ *
+ * Although the application eventually overwrites every chunk completely the library has know way of knowing this before
```
Comment (derobins) [inline doxygen/dox/DSChunkingIssues.dox]: "no way", not "know way"

File: doxygen/dox/DSChunkingIssues.dox [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+ * hand since most calls to #H5Dwrite modify only a portion of any given chunk. Therefore, the first modification of a
```
Comment (derobins) [inline doxygen/dox/DSChunkingIssues.dox]: beforehand is one word

File: doxygen/dox/DSChunkingIssues.dox [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+ * chunk will cause the chunk to be read from disk into the chunk buffer through the filter pipeline. Eventually HDF5 might
+ * contain a data set transfer property that can turn off this read operation resulting in write efficiency which is equal
```
Comment (derobins) [inline doxygen/dox/DSChunkingIssues.dox]: dataset

File: doxygen/dox/DSChunkingIssues.dox [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+ *
+ * By default the strip size is 1MB but it can be changed by calling #H5Pset_buffer.
+ *
+ * \section sec_hdf5_chunk_issues_store File Storage Overhead
+ * The chunks of the dataset are allocated at independent locations throughout the HDF5 file and a B-tree maps chunk
+ * N-dimensional addresses to file addresses. The more chunks that are allocated for a dataset the larger the B-tree.
+ *
+ * Large B-trees have two disadvantages:
+ * \li The file storage overhead is higher and more disk I/O is required to traverse the tree from root to leaves.
+ * \li The increased number of B-tree nodes will result in higher contention for the meta data cache.
```
Comment (derobins) [inline doxygen/dox/DSChunkingIssues.dox]: metadata, not meta data

**Comment by derobins** [comment]
it's nice to move this into the UG. We're going to need to revisit this text in the future, since a lot of it was clearly written before HDF5 1.10 and is a little obsolete.

**Review by derobins — APPROVED** [review]

**Pull request merged**